### PR TITLE
[FIX] travis_weblate.py: Fix the python3 when the script travis_weblate.py run

### PR DESCRIPTION
**Pull request by JesusZapata**
Fix the `python3` when the script `travis_weblate.py` run

### CODE REVIEW

**Comment by JesusZapata** [comment]
@moylop260 
With this change now the script travis_weblate.py ran well over `python2` and `python3`

Python 2

image

Python 3

image

File: travis/odoo_connection.py [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
 from contextlib import closing
-try:
-    from StringIO import StringIO
-except ImportError:
-    from io import StringIO
+
+if sys.version_info > (3,):
+    from io import BytesIO as Buffer
+else:
+    from StringIO import StringIO as Buffer
```
Comment (moylop260) [inline travis/odoo_connection.py]: why this change is required?

File: travis/git_run.py [hunk 2]
```diff
@@ -22,6 +22,8 @@ def run(self, command):
             res = subprocess.check_output(cmd)
         except subprocess.CalledProcessError:
             res = None
+        if isinstance(res, bytes):
+            res = res.decode()
```
Comment (moylop260) [inline travis/git_run.py]: what about use all in bytes?

Using res.strip(b’\n’) and so on

Comment (JesusZapata) [reply travis/git_run.py]: I think this change is less invasive because the method `git.run` is executed many times

Are you agree?

File: travis/odoo_connection.py (hunk 1 above)
Comment (JesusZapata) [reply travis/odoo_connection.py]: I try to let the previous code but this code fail when the `travis_weblate.py` run over `python3`

```python
Traceback (most recent call last):
  File "/home/odoo/maintainer-quality-tools/travis/travis_weblate.py", line 264, in <module>
    main()
  File "/home/odoo/maintainer-quality-tools/travis/travis_weblate.py", line 257, in main
    TravisWeblateUpdate().update()
  File "/home/odoo/maintainer-quality-tools/travis/travis_weblate.py", line 228, in update
    if self._generate_odoo_po_files(component['name']):
  File "/home/odoo/maintainer-quality-tools/travis/travis_weblate.py", line 131, in _generate_odoo_po_files
    new_content = odoo_context.get_pot_contents(module, lang)
  File "/home/odoo/maintainer-quality-tools/travis/odoo_connection.py", line 55, in get_pot_contents
    self.trans_export(lang, [addon], buf, 'po', self.cr)
  File "/home/odoo/odoo-11.0/odoo/tools/translate.py", line 713, in trans_export
    _process(format, modules, translations, buffer, lang)
  File "/home/odoo/odoo-11.0/odoo/tools/translate.py", line 669, in _process
    writer.write_infos(modules)
  File "/home/odoo/odoo-11.0/odoo/tools/translate.py", line 622, in write_infos
    'now': datetime.utcnow().strftime('%Y-%m-%d %H:%M')+"+0000",
  File "/.repo_requirements/virtualenv/python3.5/lib/python3.5/codecs.py", line 718, in write
    return self.writer.write(data)
  File "/.repo_requirements/virtualenv/python3.5/lib/python3.5/codecs.py", line 377, in write
    self.stream.write(data)
TypeError: string argument expected, got 'bytes'
```
I found in the odoo code the way to export the language and in this code is used the `io.BytesIO` instead of `io.StringIO()`

```python
with contextlib.closing(io.BytesIO()) as buf:
    tools.trans_export(lang, mods, buf, this.format, self._cr)
    out = base64.encodestring(buf.getvalue())
```
https://github.com/odoo/odoo/blob/cd48a806ac636806130efcf5c055f73758cb0902/odoo/addons/base/module/wizard/base_export_language.py#L37-L39

Comment (moylop260) [reply travis/odoo_connection.py]: You are talking about use `BytesIO` instead of use `StringIO` that is fine.
But you are changing from `try-except` to `if`
I think you can use try-except using BytesIO too
